docs/sphinx/manual/creating_hpx_projects.rst

```diff
@@ -230,8 +230,66 @@ Additionally, if you wish to require |hpx| for your project, replace the
 
 You can check if |hpx| was successfully found with the ``HPX_FOUND`` CMake variable.
 
-The simplest way to add the |hpx| component is to use the ``add_hpx_component``
-macro and add it to the ``CMakeLists.txt`` file:
+Using |hpx| targets
+-------------------
+
+The recommended way of setting up your targets to use |hpx| is to link to the
+``HPX::hpx`` target:
+
+.. code-block:: cmake
+
+   target_link_libraries(hello_world_component PUBLIC HPX::hpx)
+
+This requires that you have already created the target like this:
+
+.. code-block:: cmake
+
+   add_library(hello_world_component SHARED hello_world_component.cpp)
+   target_include_directories(hello_world_component PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+When you link your library to ``HPX::hpx`` you will be able use |hpx|
+functionality in your library. To create a component, however, requires setting
+two additional compile definitions:
+
+.. code-block:: cmake
+
+   target_compile_options(hello_world_component
+     HPX_COMPONENT_NAME=hello_world
+     HPX_COMPONENT_EXPORTS)
+
+Instead of setting these definitions manually you may link to the
+``HPX::component`` target, which sets ``HPX_COMPONENT_NAME`` to
+``hpx_<target_name>``, where ``<target_name>`` is the target name of your
+library. Note that these definitions should be ``PRIVATE`` to make sure these
+definitions are not propagated transitively to dependent targets.
+
+In addition to making your library a component you can make it a plugin. To do
+so link to the ``HPX::plugin`` target. Similarly to ``HPX::component`` this will
+set ``HPX_PLUGIN_NAME`` to ``hpx_<target_name>``. This definition should also be
+``PRIVATE``. Unlike regular shared libraries, plugins are loaded at runtime from
+certain directories and will not be found without additional configuration.
+Plugins should be installed into a directory containing only plugins. For
+example, the plugins created by |hpx| itself are installed into the ``hpx``
+subdirectory in the library install directory (typically ``lib`` or ``lib64``).
+When using the ``HPX::plugin`` target you need to install your plugins into an
+appropriate directory. You may also want to set the location of your plugin in
+the build directory with the ``*_OUTPUT_DIRECTORY*`` CMake target properties to
+be able to load the plugins in the build directory. Once you've set the install
+or output directory of your plugin you need to tell your executable where to
+find it at runtime. You can do this either by setting the environment variable
+``HPX_COMPONENT_PATHS`` or the ini setting ``hpx.component_paths`` (see
+:option:`--hpx:ini`) to the directory containing your plugin.
```

Comment:
@hkaiser your eyes on this paragraph would be greatly appreciated. I didn't try this myself, but this is what I understood from the source. The last part on `HPX_COMPONENT_PATHS`/`hpx.component_paths` is especially important.

Reply:
I will go back to the sources for that. I only remember that we did change things there about a year ago...

Reply:
The description looks about right. Sorry it took so long...